.circleci/config.yml

```diff
@@ -291,6 +298,20 @@ commands:
           name: Wait for local UI server to start
           command: dockerize -wait tcp://localhost:4200 -timeout 4m
 
+  skip_e2e_commit:
+    description: "Halt job if commit message contains 'SKIP E2E'"
+    steps:
+      - run:
+          name: "Halt job if commit message contains 'SKIP E2E'"
```

Comment:
On the subject of the "skip e2e" behavior - I agree with this approach, that the tests should be run by default.

Reply:
Cool.

Reply:
aside:  Eventually it might be nice to encapsulate all the trigger evaluation and job management into  a single "driver" task, and have separate configs for the real  work.

It might also be  worth looking at a small e2e smoke test suite that runs quickly and still gives reasonable integration coverage. Unless managing the environment is a  huge contributor to the time.